chalk-engine/src/forest.rs

```diff
@@ -213,11 +201,15 @@ impl<'me, C: Context, CO: ContextOps<C>> AnswerStream<C> for ForestSolver<'me, C
                 }
 
                 Err(RootSearchFail::QuantumExceeded) => {}
+
+                Err(RootSearchFail::NegativeCycle) => {
+                    panic!("negative cycle was detected");
```

Comment:
```suggestion
                    // Negative cycles *ought* to be avoided by construction. Hence panic
                    // if we find one, as that likely indicates a problem in the chalk-solve
                    // lowering rules. (In principle, we could propagate this error out,
                    // and let chalk-solve do the asserting, but that seemed like it would
                    // complicate the function signature more than it's worth.)
                    panic!("negative cycle was detected");
```